ansible_base/lib/utils/encryption.py

```diff
@@ -43,13 +43,33 @@ def __init__(self):
         self._encryption_key = self.key[32:]
         self._backend = default_backend()
 
+    def is_encrypted_string(self, value:str) -> bool:
+        # Ensure input is a string already encrypted by our algorithm
+        encrypt_sequence = value.split('$')
+
+        if len(encrypt_sequence) != 5:
+            return False
+        
+        _, flag, encoding, algo, b64data = value.split('$')
```

Comment:
This is going to assume that `ENCRYPTED_STRING` is always going to be `$encrypted$`; which it really should be but you never know when someone might be like "we need to change this". If we change this up a bit, I think we can protect ourselves incase that day ever comes. My inclination would be to move some of the starting code from `decrypt_string` into here:
```
        # This is needed because we don't enforce type hints at runtime, only in tests
        if type(value) is not str:
            raise ValueError("decrypt_string can only accept string")

        if not value.startswith(ENCRYPTED_STRING):
            return False
        
        encoding, algo, b64data = value[len(ENCRYPTED_STRING) :].split('$')
```

Then we can have `decrypt_string` do something like:
```
    def decrypt_string(self, value: str) -> str:
        if not self.is_encrypted_string(value):
              return value
        .... do the rest of the decryption ....
```

Reply:
thank you for pointing this out.
At first I thought the string deconstruction part in decrypt_string() was lengthy. However, with your insight, I now understand the rationale behind each step